include/LightGBM/utils/openmp_wrapper.h

```diff
@@ -17,20 +17,17 @@
 #include <stdexcept>
 #include <vector>
 
+static int lgb_global_omp_threads = omp_get_max_threads();
```

Comment:
I've been testing an approach like this over on my fork (https://github.com/jameslamb/LightGBM/pull/55) and I don't think just having `static int` in a header here will work.

I found that when I did that, everything that included `openmp_wrapper.h` got its *own* copy of that integer, and that therefore updates like this:

https://github.com/microsoft/LightGBM/blob/8ed371cee49cf86740b25dd9a4b985a75c9f2dba/src/c_api.cpp#L316

did not propagate through everywhere.